### PR TITLE
Add propertybag to parsed tsconfig.json file.

### PR DESCRIPTION
**Pull request by paulvanbrenk**

### CODE REVIEW

**Comment by mhegazy** [comment]
so what is the context for this change? and can we do the same for typingsOptions as well?

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -664,6 +664,22 @@ namespace ts {
...
+
+        // Contains the properties on the json we don't recognize, but the
+        // host might so we return them as a property bag. This allows the host to handle
+        // them, but doesn't have to deal with removing comments from the source json.
+        const other: any = {};
+
+        const knownProperties = ["compilerOptions", "typingOptions", "files", "exclude"];
+
+        for (const prop in json) {
+            if (knownProperties.indexOf(prop) >= 0) {
```
Comment (vladima) [inline src/compiler/commandLineParser.ts]: `ts.indexOf`

**Comment by paulvanbrenk** [comment]
@mhegazy This is to make the handling of `compileOnSave` more reliable in VS. I assume we wanted to make `typingOptions` a first class citizen in the VS code.

File: src/compiler/commandLineParser.ts (hunk 1 above)
Comment (paulvanbrenk) [inline src/compiler/commandLineParser.ts]: Done.

**Comment by mhegazy** [comment]
how about "raw" instead of other, and that has a copy of the tsconfig.json parsed content?

File: src/compiler/types.ts [hunk 2]
```diff
@@ -2592,6 +2592,7 @@ namespace ts {
         options: CompilerOptions;
         typingOptions?: TypingOptions;
         fileNames: string[];
+        raw?: any;
```
Comment (vladima) [inline src/compiler/types.ts]: why it is optional if you always set it?

File: src/services/shims.ts [hunk 3]
```diff
@@ -997,6 +997,7 @@ namespace ts {
                             options: {},
                             typingOptions: {},
                             files: [],
+                            other: {},
```
Comment (vladima) [inline src/services/shims.ts]: `raw`?

**Comment by vladima** [comment]
:+1: after comments are addressed

File: src/compiler/types.ts (hunk 2 above)
Comment (paulvanbrenk) [inline src/compiler/types.ts]: good point

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**